core/src/test/java/org/apache/calcite/sql/parser/SqlParserUtilTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.parser;
+
+import org.apache.calcite.avatica.util.TimeUnit;
+import org.apache.calcite.sql.SqlIntervalQualifier;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Tests {@link SqlParserUtil}.
+ */
+public class SqlParserUtilTest {
```

Comment:
SqlParserIntervalQualifierTest?

Reply:
I figured that at some point, this test file might be expanded to include more tests for `SqlParserUtil`, so I named it after the main class itself. However, please let me know if it's considered better to scope the test files more narrowly.

Reply:
It seems that the impact is not great.
I think we can add JavaDoc to let subsequent contributors know what this test file is for.

Reply:
I've added some additional javadoc.